### PR TITLE
[Server] Format deployment events

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — COMMENTED** [review]
@MUzairS15 there are a couple items (lint checks) to review - https://github.com/meshery/meshery/actions/runs/9351539728/job/25737570183?pr=11102#step:4:29

**Comment by MUzairS15** [comment]
Yes, I will be fixing those as I complete the PR.
There’s a small update required.

File: server/models/pattern/patterns/patterns.go [hunk 1]
```diff
@@ -18,7 +17,21 @@ import (
...
+	Kind       string
+	CompName   string
+	Success    bool
+	DesignName string
+	Message    string
+}
+
+type DeploymentMessagePerContext struct {
+	Summary    []DeploymentMessagePerComp
+	ServerName string
```
Comment (leecalcote) [inline server/models/pattern/patterns/patterns.go:30]: Is "ServerName" defined in the Events schema?

File: server/models/pattern/patterns/patterns.go [hunk 2]
```diff
@@ -18,7 +17,21 @@ import (
 	"github.com/spf13/viper"
 )
 
-func ProcessOAM(kconfigs []string, oamComps []string, oamConfig string, isDel bool, patternName string, ec *models.Broadcast, userID string, provider models.Provider, hostname v1beta1.IHost, skipCrdAndOperator bool) (string, error) {
+type DeploymentMessagePerComp struct {
```
Comment (leecalcote) [inline server/models/pattern/patterns/patterns.go:20]: Is this defined as a subset of the Events schema and versioned?

Comment (leecalcote) [reply server/models/pattern/patterns/patterns.go:20]: Let's be schema-driven.

File: server/models/pattern/patterns/patterns.go [hunk 3]
```diff
@@ -18,7 +17,21 @@ import (
...
+	Message    string
+}
+
+type DeploymentMessagePerContext struct {
+	Summary    []DeploymentMessagePerComp
+	ServerName string
+	Location   string
+}
+
+func ProcessOAM(kconfigs []string, oamComps []string, oamConfig string, isDel bool, patternName string, ec *models.Broadcast, userID string, provider models.Provider, hostname v1beta1.IHost, skipCrdAndOperator bool) ([]DeploymentMessagePerContext, error) {
```
Comment (leecalcote) [inline server/models/pattern/patterns/patterns.go:34]: Do we still need a function with OAM in the name?

**Comment by leecalcote** [comment]
Merge conflict.

File: server/models/pattern/patterns/patterns.go (hunk 1 above)
Comment (MUzairS15) [reply server/models/pattern/patterns/patterns.go]: resolved in slack,

**Pull request merged**